orchestrator-salt/src/main/resources/salt-common/salt/filecollector/template/cloud_storage_upload.sh.j2

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+{%- from 'filecollector/settings.sls' import filecollector with context %}
+{%- from 'telemetry/settings.sls' import telemetry with context %}
+
+function log() {
+  echo "$1"
+}
+
+function error_exit() {
+  log "ERROR $1"
+  exit 1
+}
+
+function upload_file() {
+   local platform="$2"
+   if [[ "$platform" == "AWS" ]]; then
+      upload_to_s3 "$1"
+   elif [[ "$platform" == "AZURE" ]]; then
+      upload_to_azure_storage "$1"
+   else
+     log "Upload file to cloud storage is not supported from platform '${platform}'"
+   fi
+}
+
+function upload_to_s3() {
+  local filename=$1
+  s3_filename="$(basename -- $filename)"
+  s3_file_path="{{ filecollector.s3BaseUrl }}/${s3_filename}"
+  # --sse AES256
+  /usr/bin/aws s3 cp --no-progress "${filename}" "${s3_file_path}"
```

Comment:
We will need to deal with encryption + the region must be a parameter otherwise it won't work with S3 VPC endpoint

Reply:
i have mentioned in the PR, encryption is not yet handled. will be done later, no UI or tests around that so that can be done later. btw, I will work on a binary lib that contains these commands with enhanced options, so that won't be final anyway (as we will have a lot of different scripts, it's better to put/use them together)

Reply:
What about VPC endpoints? Do we have a jira for that as well? I'm fine if we address that later, but let's have a jira so we don't forget.